pysparkling/tests/test_resolve_filenames.py

```diff
@@ -33,7 +56,36 @@ def test_s3_1():
             'crawl-data/CC-MAIN-2015-11/warc.paths.gz' in filenames)
 
 
+def test_hdfs_resolve_filenames():
+    from pysparkling.fileio.fs import Hdfs
+    unused_path = "/"
+    path = "hdfs://hdfs-cluster.com/user/username/input/part-*.gz"
```

Comment:
Can you add a test for a path without a `*` or `?`?
And another test for omitting the `/part-*.gz`?

Reply:
Adding tests without `/part-*.gz` allowed me to notice a misalignment with pyspark which support both `hdfs://hdfs-cluster.com/user/username/input/` and `hdfs://hdfs-cluster.com/user/username/input`,

I modified the code and tests but it increase a bit the complexity

It still does not support `hdfs://hdfs-cluster.com/user/username/*/part-*.gz` but I believe that that should be another PR